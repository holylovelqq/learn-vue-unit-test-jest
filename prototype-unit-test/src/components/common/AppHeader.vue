<template>
    <div class="app-header">
        <el-menu
            class="elmenu"
            :default-active="activeIndex"
            mode="horizontal"
            @select="handleSelect"
            background-color="#333"
            text-color="#fff"
            active-text-color="#fff">
            <router-link :to="{ name: 'Incidents' }">
              <el-menu-item class="elmenuitem" index="1">Home</el-menu-item>
            </router-link>
        </el-menu>
    </div>
</template>

<script>
export default {
  data () {
    return {
      activeIndex: '1',
      testKey: 1
    }
  },
  methods: {
    handleSelect (key, keyPath) {
      // console.log(key, keyPath)
      this.testKey = 5
    }
  }
}
</script>

<style>
</style>
