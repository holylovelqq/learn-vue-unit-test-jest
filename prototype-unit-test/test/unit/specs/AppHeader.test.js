import { shallowMount, createLocalVue } from '@vue/test-utils'
import Router from 'vue-router'
import elementUI from 'element-ui'
import AppHeader from '@/components/common/AppHeader'

const localVue = createLocalVue()
localVue.use(elementUI)
localVue.use(Router)

describe('Appheader.vue', () => {
  const wrapper = shallowMount(AppHeader, {localVue})
  it('AppHeader DOM test', () => {
    expect(wrapper.is(AppHeader)).toBeTruthy()
    expect(wrapper.find('.elmenu').exists()).toBe(true)
    expect(wrapper.find('.elmenuitem').text()).toBe('Home')
  })

  it('appheader handleSelect test', () => {
    wrapper.vm.handleSelect()
    expect(wrapper.vm.testKey).toBe(5)
  })

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
