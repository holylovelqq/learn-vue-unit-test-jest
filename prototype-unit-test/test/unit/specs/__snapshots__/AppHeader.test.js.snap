// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appheader.vue matches snapshot 1`] = `
<div class="app-header">
    <elmenu-stub mode="horizontal" defaultactive="1" menutrigger="hover" backgroundcolor="#333" textcolor="#fff" activetextcolor="#fff" collapsetransition="true" class="elmenu">
        <router-link-stub to="[object Object]" tag="a" event="click">
            <elmenuitem-stub index="1" class="elmenuitem">Home</elmenuitem-stub>
        </router-link-stub>
    </elmenu-stub>
</div>
`;
